File: FixTwoNodes.java
class Solution {
    Node prev = null;
    Node first = null;
    Node sec = null;
    void correctBST(Node root) {
        // code here.
        inorderTraversal(root);
        if(first != null && sec != null){
            int temp = first.data;
            first.data = sec.data;
            sec.data = temp;
        }
    }
    
    void inorderTraversal(Node root){
        if(root == null){
            return;
        }
        
        inorderTraversal(root.left);
        if(prev != null && root.data < prev.data){
            if(first == null){
                first = prev;
            }
            
            sec = root;
        }
        prev = root;
        inorderTraversal(root.right);
    }
}

File: FixTwoNodes.cpp
class Solution {
  public:
    Node* prev = nullptr;
    Node* first = nullptr;
    Node* second = nullptr;

    void correctBST(Node* root) {
        inorderTraverse(root);
        if (first && second) {
            swap(first->data, second->data);
        }
    }

private:
    void inorderTraverse(Node* root) {
        if (!root) return;

        inorderTraverse(root->left);

        if (prev && root->data < prev->data) {
            if (!first) {
                first = prev;
            }
            second = root;
        }
        prev = root;

        inorderTraverse(root->right);
    }
};
